packages/node-core/src/indexer/fetch.service.ts

```diff
@@ -60,8 +60,11 @@ export abstract class BaseFetchService<DS extends BaseDataSource, B extends IBlo
   }
 
   private get latestFinalizedHeight(): number {
-    assert(this._latestFinalizedHeight, new Error('Latest Finalized Height is not available'));
-    return this._latestFinalizedHeight;
+    if (!this.nodeConfig.unfinalizedBlocks) {
+      assert(this._latestFinalizedHeight, new Error('Latest Finalized Height is not available'));
+      return this._latestFinalizedHeight;
+    }
+    return this.latestBestHeight;
```

Comment:
I think this can still prefer latest finalized height

```suggestion
    return this._latestFinalizedHeight ?? this.latestBestHeight;
```